scipy/sparse/_construct.py

```diff
@@ -507,11 +507,20 @@ def _compressed_sparse_stack(blocks, axis):
         sum_dim += b.shape[axis]
         last_indptr += b.indptr[-1]
     indptr[-1] = last_indptr
+    # TODO remove this if-structure when sparse matrices removed
+    if container is coo_matrix:
```

Comment:
Should this be `if isinstance(container, spmatrix)`?

Reply:
Or I suppose maybe `container` could be changed to a boolean `return_sparse_array` parameter, given how you're calling this. That would make this part simpler, I think.

Reply:
Alternatively, could this always return an `sparray`, then have `bmat` handle the conversion to matrix?

So `bmat` becomes something like:

```python
    result = _block_array([blocks], format, dtype)
    if any(isinstance(b, sparray) for b in blocks.flat):
        return result
    else:
        return getattr(sparse, f"{result.format}_matrix")(result)
```

Reply:
I changed the passed value to be boolean `return_spmatrix` with a default value of `False` so the array tests can remain the same when this testing removes spmatrix.